ebos/collecttoiorank.hh

```diff
@@ -708,19 +708,80 @@ public:
             }
         }
 
+    };
+
+    class PackUnPackAquiferData : public P2PCommunicatorType::DataHandleInterface
+    {
+        const Opm::data::Aquifers& localAquiferData_;
+        data::Aquifers& globalAquiferData_;
+
+    public:
+        PackUnPackAquiferData(const Opm::data::Aquifers& localAquiferData,
+                              Opm::data::Aquifers& globalAquiferData,
+                              bool isIORank)
+            : localAquiferData_(localAquiferData)
+            , globalAquiferData_(globalAquiferData)
+        {
+            if (isIORank) {
+                MessageBufferType buffer;
+                pack(0, buffer);
+
+                // pass a dummyLink to satisfy virtual class
+                int dummyLink = -1;
+                unpack(dummyLink, buffer);
+            }
+        }
+
+        // pack all data associated with link
+        void pack(int link, MessageBufferType& buffer)
+        {
+            // we should only get one link
+            if (link != 0)
+                throw std::logic_error("link in method pack is not 0 as expected");
+
+            int size = localAquiferData_.size();
+            buffer.write(size);
+            for (const auto& [key, data] : localAquiferData_) {
+              buffer.write(key);
+              data.write(buffer);
+            }
+        }
+
+        // unpack all data associated with link
+        void unpack(int /*link*/, MessageBufferType& buffer)
+        {
+            int size;
+            buffer.read(size);
+            for (int i = 0; i < size; ++i) {
+              int key;
+              buffer.read(key);
+              data::AquiferData data;
+              data.read(buffer);
+              auto& aq = globalAquiferData_[key];
+              aq.aquiferID = std::max(aq.aquiferID, data.aquiferID);
+              aq.pressure = std::max(aq.pressure, data.pressure);
+              aq.initPressure = std::max(aq.initPressure, data.initPressure);
+              aq.datumDepth = std::max(aq.datumDepth, data.datumDepth);
+              aq.fluxRate += data.fluxRate;
+              aq.volume += data.volume;
```

Comment:
This will makes the parallel restarting more difficult.  Because with this, each process only have its own portion of `volume` (theoretically), when restarting, you only have the one for the whole aquifer, then it is not easy to recover the individual potions anymore. 

I suggest we handle the parallel summation within the simulation code (aquifer model code), then all the processes have the same `volume`. 

`fluxRate` is probably should be the same since we use `fluxRate` to update `volume`.

Reply:
I am not sure you are aware that this not connected directly to restart.
Unpackinging will be done on the root process during file output only. It will unpack one result for each process. The result is only written to disk. Hence only the root process needs to have the same result as a serial run (I guess that is assured by taking the max or the sum).

Reply:
> I am not sure you are aware that this not connected directly to restart.

It is not directly related to restart. But when considering restarting, how you can manage to get some meaningful `volume` for each process to begin with, because `volume` is an accumulative term.  Although at the end, it is the root process result that matters for output, but imagining you have some meaningful information on the root process, and other processes only have some half meaningful information after restarting. I do not think that is pretty.

Reply:
For what it's worth, the restart file will have certain pieces of information pertaining to each aquifer-reservoir connection in addition to values aggregated to the aquifer as a whole. I haven't fully characterised everything yet. It **may** happen that we need to maintain additional dynamic quantities at the connection level.